### ISSUE TITLE
Filter location 'virtual' or 'online'

### ISSUE THREAD

**Issue by Binnette** [thomersch/openstreetmap-calendar#44]
Hi, is it possible to add a location filter for value 'virtual/online' event?

Also, we may consider in a new event to set a location, for example Paris in France and check the box "online". That way local people AND remote people can join the event both IRL and online?

Have a nice day.

**Comment by thomersch** [comment]
Thanks for reporting.

I've been thinking about it, but haven't reached any conclusion, yet. I see that there are events which have local relevance, but currently take place on some kind of online medium.

One of the issues I see is: What about events that do have a physical location, but have some kind of streaming/online participation? Those would be “online”, but not mainly/exclusively, which could be confusing to participants reading it.

I understand that this is a feature which is pretty important _now_, but maybe won't be in a year, but we will have to keep it for eternity.

**Comment by Binnette** [comment]
Hum may be, we should consider at least adding a checkbox "Online event: yes/no" when creating an event. To simply, display an "online" in the event header or an icon 🌐 ?

**Comment by jbelien** [comment]
> Event organizers usually write something like "virtual" or "online" in the event title anyway, which is good enough for now in my opinion.

Indeed, for OSMBE, we put "virtual" or "online" in the title and set "Online" as location (see https://osmcal.org/event/284/)

**Comment by sfkeller** [comment]
> organizers usually write something like "virtual" or "online" in the event title anyway, which is good enough for now in my opinion. 

I fear it's not good enough.

Not all events are having "virtual/online" in the title. 

It's contradicting (and misleading) if in the title there's "virtual/online" while the location is set. See for example this event https://osmcal.org/event/425/ and the fact that it's now announced in Weggis here in Weekly https://weeklyosm.eu/de/archives/14443 .

So, an separate attribute is needed and shown in GUI (as checkbox "Online event: no (default: no)").

How you encode this attribute in the API is up to you. An additional attribute is IMHO not such a breaking change. But think that you already offered RSS with parameter events.rss?in=Germany  (which is OK if there's an additional attribute).

**Comment by natrius** [comment]
Putting "virtual" in the title is nice, but the location is still needed. It would be nice to be able to just put in "Austria" or "Germany" as location and nothing more detailed. Or "Salzburg", "Oberösterreich" or "Bayern" or "Berlin" or "Bremen". This way its not a specific point on a map and though the title its still clear its a virtual one.

**Comment by natrius** [comment]
It is still needed, in my opinion. We are still holding national online events additionally to local offline ones so its possible to contact and talk with other you normally would not. It would be nice - as said - to be able to just use 'Austria' in the location field, that would help already. That might be an easy workaround and might help with it? Currently it looks like 

![grafik](https://user-images.githubusercontent.com/7243447/189642105-b55b4d58-046f-4542-8ad7-6980835b7868.png)

That where the information "Großarl, Salzburg" is absolutely not needed and even might be misleading.

**Comment by brierjon** [comment]
I'm also struggling with the accurate representation for our virtual meeting - https://osmcal.org/event/2187/. Virtual in the title isn't sufficient as it does not cover the format which is separate from type and location (ie applicable area of interest for virtual rather than location of the meeting). Format (in person, hybrid (both in person and online, virtual (digital only)). 

For us, the location would be applicable to the area of interest the event should be targeted rather than the physical location of the meeting. The current work around is place the marker in a unlikely location for a physical meetup so the event will be captured and classified correctly for the area of interest.

I suggest adding:
Format (in person, hybrid, virtual)
The location field needing to be conditional to format.
* in person - a point
* hybrid - needing both the ability to mark the physical location(s) as well as possibly indicate the broader area of interest the event is targeting for broader for reach and accurate metadata
* virtual - just needing the area of targeted recruitment to be available which could leverage relations, ways etc to show the virtual recruitment allowing the event to populate in the respective geographically targeted area. (similar to natrius describes in https://github.com/thomersch/openstreetmap-calendar/issues/44#issuecomment-1243602206

**Comment by natrius** [comment]
I would love to see simply the relation, for example. https://www.openstreetmap.org/relation/35183 or for the event https://osmcal.org/event/2778/ simply https://www.openstreetmap.org/relation/62369 that instead of a point on the Sacrow-Paretzer Kanal :)
